packages/go_router/lib/src/builder.dart

```diff
@@ -50,6 +50,11 @@ class RouteBuilder {
 
   final GoRouterStateRegistry _registry = GoRouterStateRegistry();
 
+  /// Caches a HeroController for the nested Navigator, which solves cases where the
+  /// Hero Widget animation stops working when navigating.
+  final Map<GlobalKey<NavigatorState>, HeroController> _goHeroCache =
```

Comment:
At the end of the build method, we need to clean up the cache that doesn't have associated navigatorKey to prevent memory leak

Reply:
@chunhtai, it seems that cache clean-up logic was added. What else do we need to improve before marge?